src/Symfony/Component/Notifier/Bridge/Pusher/PusherTransport.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Pusher;
+
+use Pusher\Pusher;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\PushMessage;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Throwable;
+
+/**
+ * @author Yasmany Cubela Medina <yasmanycm@gmail.com>
+ */
+final class PusherTransport extends AbstractTransport
+{
+    private $pusherClient;
+
+    public function __construct(Pusher $pusherClient, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->pusherClient = $pusherClient;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        $settings = $this->pusherClient->getSettings();
+        preg_match('/api-([\w]+)\.pusher\.com$/m', $settings['host'], $server);
+
+        return sprintf('pusher://%s:%s@%s?server=%s', $settings['auth_key'], $settings['secret'], $settings['app_id'], $server[1]);
```

Comment:
This method should not return sensitive information like the secret. The goal is not to reconstruct a valid DSN, but to give the main configuration information. See other bridges for some examples.